### DISCUSSION TITLE
Memory exhausted

### DISCUSSION THREAD

**Discussion by bluewhiskeywhisper** [rappasoft/laravel-livewire-tables#1356] [Q&A] [answered]
Hi there,

This is a great project and I've enjoyed playing with it over the weekend! Thanks for all the hard work building and maintaining it.

Question: I have a table with 500k+ rows and 20+ columns, and I'm hitting `Allowed memory size exhausted` exception. Of course I could increase the PHP memory but I don't understand why so much memory is needed. Are you able to explain why? I thought pagination would limit the amount of content trying to be rendered. I still get the same issue when my builder limits it to just one result. Here's my `builder()`:

`public function builder(): Builder
 {
  return Message::query()->where('id', 1);
 }`

I also have another table with a 100+ rows and 5+ columns, and it works really well.

**Using:** `dev-v3-master` (4250511), PHP `8.2.10`, and Laveral `10.23.1`.

**Comment by lrljoe** [comment]
Did you find the solution to your issue, if so, please can you share it here so that others can benefit!

I have a demo environment for large table testing, so happy to try and replicate any issues, and its what I use to review optimisation potential.

**Reply by bluewhiskeywhisper** [reply, marked as answer]
Hi @lrljoe, yes, sorry, I did. It was entirely unrelated though. I had a bug in my code left over from testing where I was passing in the whole content 500k+ rows as an array into a function as a variable. I got rid of that and it now works fine.

**Reply by lrljoe** [reply]
Oh wow,  don't forget the options to serialise models for working with them, and only pulling back specific fields/not loading unnecessary relations (which this package handles for you for your table columns)

there are some general laravel/livewir performance tips on the sponsors only content that I found really useful btw, and there's more to come ( I do not benefit from any sponsorship btw! )

Thanks for raising it anyway, and do feel free to drop in on the discord, we are a friendly and helpful bunch.
